File: OOP-Java/output/com/output27/Car.java
package com.output27;

public class Car extends Vehicle {
	int numGears;
	boolean isConvertible;

	public Car(int numGears, int maxSpeed) {
		super(maxSpeed);
		this.numGears = numGears;
		this.isConvertible = true;
	}

	public boolean isConvertible() {
		return isConvertible;
	}
}

File: OOP-C++/Part3/Basics/1 Inheritance/Private Inheritance/Car.cpp
class Car : private Vehicle {

  public:
    int numGears;

    void print() {
        cout << "NumTyres : " << numTyres << endl; // Can access protected property as it is sub-class of Vehicle.
        cout << "Color : " << color << endl;       // Can access public property of Vehicle
        cout << "Num gears : " << numGears << endl;
        //cout<<"Max Speed : " << maxSpeed<<endl;		//maxSpeed is private within context
    }
};

/*
   If we didn't specify access modifier while inheriting by default access modifier
   is private.
 */
